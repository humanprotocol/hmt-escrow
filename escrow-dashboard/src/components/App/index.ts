export { App } from './App';
export { AppNetworkContext } from './AppNetworkContext';
