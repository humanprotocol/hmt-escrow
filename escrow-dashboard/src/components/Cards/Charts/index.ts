import CardBarChart from './BarChart';
import CardStackedBarChart from './StackedBarChart';

export { CardBarChart, CardStackedBarChart };
