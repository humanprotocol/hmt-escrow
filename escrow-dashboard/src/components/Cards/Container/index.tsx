import Box from '@mui/material/Box';
import React from 'react';

export const CardContainer = ({ children }: { children: React.ReactNode }) => {
  return (
    <Box
      sx={{
        background: '#fff',
        boxShadow:
          '0px 3px 1px -2px #E9EBFA, 0px 2px 2px rgba(233, 235, 250, 0.5), 0px 1px 5px rgba(233, 235, 250, 0.2)',
        borderRadius: { xs: '8px', xl: '16px' },
        padding: { xs: '21px 24px 12px', xl: '54px 60px 32px' },
      }}
    >
      {children}
    </Box>
  );
};
