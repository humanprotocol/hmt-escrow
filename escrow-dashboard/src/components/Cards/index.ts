export * from './Charts';
export { CardTextBlock } from './TextBlock';
