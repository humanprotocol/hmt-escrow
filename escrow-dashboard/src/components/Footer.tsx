import * as React from 'react';
import Box from '@mui/material/Box';
import { Stack, Typography } from '@mui/material';

import smallLogoSvg from 'src/assets/small-logo.svg';
import GithubIcon from './Icons/GithubIcon';
import DiscordIcon from './Icons/DiscordIcon';
import TwitterIcon from './Icons/TwitterIcon';
import TelegramIcon from './Icons/TelegramIcon';
import LinkedinIcon from './Icons/LinkedinIcon';

const Footer: React.FC = (): React.ReactElement => {
  return (
    <Box
      sx={{
        px: 12,
        pt: '12px',
        pb: '32px',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        flexWrap: 'wrap',
      }}
    >
      <Box sx={{ display: 'flex', alignItems: 'center' }}>
        <img src={smallLogoSvg} alt="logo" />
        <Typography
          color="text.secondary"
          variant="caption"
          ml={2.5}
          lineHeight={1}
        >
          Terms and conditions
        </Typography>
      </Box>
      <Typography color="text.secondary" variant="caption">
        © {new Date().getFullYear()} HPF. HUMAN Protocol® is a registered
        trademark
      </Typography>
      <Stack direction="row" spacing={4}>
        <GithubIcon />
        <DiscordIcon />
        <TwitterIcon />
        <TelegramIcon />
        <LinkedinIcon />
      </Stack>
    </Box>
  );
};

export default Footer;
