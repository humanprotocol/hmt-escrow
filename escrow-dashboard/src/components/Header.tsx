import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';
import Toolbar from '@mui/material/Toolbar';

import humanLogo from 'src/assets/logo.svg';
import TokenPriceIndicator from './TokenPriceIndicator';

const Header: React.FC = (): React.ReactElement => (
  <Box sx={{ flexGrow: 1 }}>
    <AppBar position="fixed" sx={{ background: '#fff', boxShadow: 'none' }}>
      <Toolbar disableGutters>
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
            width: '100%',
            padding: {
              xs: '30px 8px 26px',
              sm: '30px 28px 26px',
              md: '30px 52px 26px',
            },
          }}
        >
          <Box display="flex" alignItems="center">
            <img src={humanLogo} alt="logo" />
            <Typography variant="h6" color="primary" ml="10px">
              Scan
            </Typography>
          </Box>
          <Box display="flex" alignItems="center" ml="auto">
            <TokenPriceIndicator />
            <Link
              href="https://humanprotocol.org"
              target="_blank"
              sx={{
                textDecoration: 'none',
                ml: '28px',
                display: { xs: 'none', sm: 'block' },
              }}
            >
              <Typography variant="body2" fontWeight={600}>
                HUMAN Website
              </Typography>
            </Link>
          </Box>
        </Box>
      </Toolbar>
    </AppBar>
  </Box>
);

export default Header;
