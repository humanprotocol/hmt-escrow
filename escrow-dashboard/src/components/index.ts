export { EscrowContainer as Escrow } from './Escrow';
export { TokenContainer as Token } from './Token';
export { App, AppNetworkContext } from './App';
