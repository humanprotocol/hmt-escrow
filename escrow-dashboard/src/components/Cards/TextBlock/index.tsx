import { Typography } from '@mui/material';
import numeral from 'numeral';
import React from 'react';

import { CardContainer } from '../Container';

type CardTextBlockProps = {
  title: string;
  value?: number | string;
  format?: string;
};

export const CardTextBlock: React.FC<CardTextBlockProps> = ({
  title,
  value,
  format = '0,0',
}): React.ReactElement => {
  return (
    <CardContainer>
      <Typography variant="body2" color="primary" fontWeight={600} mb="4px">
        {title}
      </Typography>
      <Typography
        variant="h2"
        color="primary"
        sx={{ fontSize: { xs: 32, md: 48, lg: 64, xl: 80 } }}
      >
        {numeral(value).format(format)}
      </Typography>
    </CardContainer>
  );
};
