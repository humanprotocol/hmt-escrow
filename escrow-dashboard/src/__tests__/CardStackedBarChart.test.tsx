import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { CardStackedBarChart } from 'src/components/Cards';

global.ResizeObserver = require('resize-observer-polyfill');

const mock = {
  allEscrowAmount: 100,
  pendingEventCount: 100,
  series: [
    { date: '23/09', dailyEscrowAmounts: 1, dailyPendingEvents: 1 },
    { date: '24/09', dailyEscrowAmounts: 2, dailyPendingEvents: 2 },
    { date: '25/09', dailyEscrowAmounts: 3, dailyPendingEvents: 3 },
  ],
};

describe('when rendered CardStackedBarChart component', () => {
  it('should render passed prop `allEscrowAmount`', async () => {
    await act(() => {
      render(
        <CardStackedBarChart
          series={mock.series}
          allEscrowAmount={mock.allEscrowAmount}
          pendingEventCount={mock.pendingEventCount}
        />
      );
    });
    expect(screen.findByLabelText(mock.allEscrowAmount)).toBeTruthy();
  });

  it('should render passed prop `pendingEventCount`', async () => {
    await act(() => {
      render(
        <CardStackedBarChart
          series={mock.series}
          allEscrowAmount={mock.allEscrowAmount}
          pendingEventCount={mock.pendingEventCount}
        />
      );
    });
    expect(screen.findByLabelText(mock.pendingEventCount)).toBeTruthy();
  });
});

it('CardBarChart component renders correctly, corresponds to the snapshot', () => {
  const tree = renderer
    .create(
      <CardStackedBarChart
        series={mock.series}
        allEscrowAmount={mock.allEscrowAmount}
        pendingEventCount={mock.pendingEventCount}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
