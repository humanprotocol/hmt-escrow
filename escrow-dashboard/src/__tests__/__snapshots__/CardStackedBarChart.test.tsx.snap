// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBarChart component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-1ncz59t"
>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-1vykxz6"
      >
        <div
          className="recharts-responsive-container"
          style={
            Object {
              "height": "100%",
              "maxHeight": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-tgim99-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-12 css-1k1029z-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-dapz8w-MuiTypography-root"
        >
          Amount of Escrows
        </p>
        <h2
          className="MuiTypography-root MuiTypography-h2 css-1ag1ig1-MuiTypography-root"
        >
          100
        </h2>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-12 css-1k1029z-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-t8mj2z-MuiTypography-root"
        >
          All Escrows Pending Events
        </p>
        <h2
          className="MuiTypography-root MuiTypography-h2 css-15bms1-MuiTypography-root"
        >
          100
        </h2>
      </div>
    </div>
  </div>
</div>
`;
